C-Sharp/VectorDistanceCalculation/PE11VectorDistanceCalculation/Point2D.cs

```diff
@@ -6,35 +6,49 @@
 
 namespace PE11VectorDistanceCalculation
 {
+    //Comments in this class apply for both objects
+    //Why have two objects? You can accomplish everything with one object and an extra property z here
+    //if you remove the methods from this class it can become a struct
     class Point2D : IEquatable<Point2D>
     {
+        //I like this class being immutable but no reason for the private set
         public int X { get; private set; }
         public int Y { get; private set; }
-
         public Point2D()
         {
+            //two problems with assigning X, Y here. 
+            //1. theyre value types, what is the default for the value types?
+            //2. what does this constructor accomplish?
             X = 0;
             Y = 0;
         }
 
+        //How do I set this object?
+        //Both constructors have internal assignments and no way for you to set the vector
+        //but I may not understand the assignment
         public Point2D(Random random, int min = 1, int max = 101)
         {
             X = random.Next(min, max);
             Y = random.Next(min, max);
         }
 
+        //This method doesnt belong here. This method belongs in a static utility class 
+        //Helper classes can maintain state
+        //Utility classes have static methods and only operate on parameters and return something
```

Comment:
I understand. When is it appropriate for classes to have methods and what kinds?

Reply:
I only allow override methods and very rare other cases for methods to be in objects. Keep them clean, this will save you pain in big applications.

Reply:
Follow-on question based on your recommendation: When should we use interfaces?  Interfaces are meant to be consumed by classes, and don't interfaces usually require implementation of methods, such as the built-in interfaces?

Reply:
I am a bit confused by your question. When should we use interfaces for objects? Or when do we use interfaces in general?

Reply:
The former, though I wouldn't mind learning about the latter as well.